packages/vite/src/node/watch.ts

```diff
@@ -68,6 +69,23 @@ export function resolveChokidarOptions(
     )
   }
 
+  if (!isRollupChokidar3) {
+    // If watch options is turned off, ignore watching anything, which essentially makes it noop
+    // eslint-disable-next-line eqeqeq -- null means disabled
+    if (options === null) {
+      ignored.push(() => true)
+    }
+    // Convert strings to picomatch pattern functions for compat
+    ignored = ignored.map((pattern) => {
+      if (typeof pattern === 'string') {
+        const matcher = picomatch(pattern)
+        return (path: string) => matcher(path)
+      } else {
+        return pattern
+      }
+    })
```

Comment:
`anymatch` seems to handle negate globs.
https://github.com/micromatch/anymatch/blob/cbd278e43710eaf325d2061fa11aefd127c509be/index.js#L85-L91

Probably it needs to be handled like:
https://github.com/vitejs/vite/blob/915cd7f11abec43d8f660665a01336870a202831/packages/vite/src/node/plugins/importMetaGlob.ts#L72-L80
https://github.com/vitejs/vite/blob/915cd7f11abec43d8f660665a01336870a202831/packages/vite/src/node/plugins/importMetaGlob.ts#L96-L101
But maybe it's fine as it doesn't seem to work now: https://github.com/vitejs/vite/issues/8619

Reply:
Yeah I think it might not be worth doing it for now. With the `ignored` pattern now, if one of the matchers return true there's no way to reverse that. It's implemented [here](https://github.com/paulmillr/chokidar/blob/97894d36657c8e4571bd1a07fb6e2a3f50f7019d/src/index.ts#L121-L126). So in practice negated globs don't really work.

However, since it now accepts a function, this could open up an opportunity for us to handle the matching ourselves entirely 🤔 Basically we pass a single function that we own that we process the `ignored` ourselves, skipping chokidar behaviour. Maybe for another PR though

Reply:
> With the `ignored` pattern now, if one of the matchers return true there's no way to reverse that.

Ah, true. Yeah, I agree we should leave this at least for now.